### PR TITLE
number: flip S.div argument order

### LINKED ISSUE #430: S.div argument order should match that of S.sub
Currently we have:

```haskell
sub :: FiniteNumber -> (FiniteNumber -> FiniteNumber)
sub_ :: FiniteNumber -> FiniteNumber -> FiniteNumber
```

I suggest we update [`S.div`][1] to match, giving:

```haskell
div :: FiniteNumber -> (FiniteNumber -> FiniteNumber)
div_ :: FiniteNumber -> FiniteNumber -> FiniteNumber
```

The _divide by two_ function would then be `S.div(2)` rather than `S.div(S.__, 2)`. :)

[1]: https://github.com/sanctuary-js/sanctuary/tree/v0.13.2#div

### PR DESCRIPTION
**Pull request by davidchambers**
Closes #430

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3827,24 +3827,49 @@
...
+  S.div_ =
+  def('div_',
+      {},
+      [$.FiniteNumber, $.NonZeroFiniteNumber, $.FiniteNumber],
+      div_);
 
   //# pow :: FiniteNumber -> (FiniteNumber -> FiniteNumber)
   //.
-  //. Takes a finite number `n` and returns the _`n` squared_ function.
+  //. Takes a finite number `n` and returns the _power of `n`_ function.
```
Comment (Avaq) [inline index.js]: Nit: I would commit this separately, allowing the "flip `div`" to be treated (eg. reverted) separately from this unrelated fix.

Comment (davidchambers) [inline index.js]: I agree that atomic commits are desirable. In this case I consulted the `pow` description when writing the `div` description, which led me to notice this error. I'd argue that this fix is _indirectly_ related rather than unrelated. ;)
